src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2WebSocketTests.cs

```diff
@@ -437,4 +428,134 @@ public async Task ExtendedCONNECT_AcceptAsyncStream_IsNotLimitedByMaxRequestBody
 
         await StopConnectionAsync(expectedLastStreamId: 1, ignoreNonGoAwayFrames: false);
     }
+
+    [Fact]
+    public async Task ExtendedCONNECTMethod_DoesNotProvideUsableBodyStreams()
+    {
+        await InitializeConnectionAsync(async context =>
+        {
+            var connectFeature = context.Features.Get<IHttpExtendedConnectFeature>();
+            // We could throw, but no-oping is adequate.
+            Assert.Equal(0, await context.Request.Body.ReadAsync(new byte[1]));
+            var ex = await Assert.ThrowsAsync<InvalidOperationException>(() => context.Response.Body.WriteAsync(new byte[1] { 0x00 }).AsTask());
+            Assert.Equal(CoreStrings.ConnectResponseCanNotHaveBody, ex.Message);
+            ex = await Assert.ThrowsAsync<InvalidOperationException>(() => context.Response.BodyWriter.WriteAsync(new byte[1] { 0x00 }).AsTask());
+            Assert.Equal(CoreStrings.ConnectResponseCanNotHaveBody, ex.Message);
+
+            var stream = await connectFeature.AcceptAsync();
```

Comment:
What happens when the endpoint does absolutely nothing for an extended CONNECT request? Just `httpContext => Task.CompletedTask`. Do we just produce an "empty" 200 response? Or a 500? I think it should be a 500 given the request was not successfully handled. Do we log anything about not properly handling an extended CONNECT in this case?

I know that isn't the original issue, but this got me thinking. If we are producing a 200 response this case, I guess this could be addressed in a follow up.

Reply:
I just tried this. The client gets back a 200, END_STREAM. They might get an additional RST_STREAM frame if the request side of the stream is still open.

I think this is fine. The connection is 'accepted' and immediately closed. Yes the server did this by mistake, but the client isn't left in a bad state.

I chalk this ambiguity up to 200 being a bad choice for an accept status code 😁.

Reply:
This is something I'd like to revisit based on customer feedback.